recipe/patches/0006-set-pyarrow._hotfix_installed-True.patch

```diff
@@ -0,0 +1,24 @@
+From c443ae531b9fad3157a22091d66f696c235913ad Mon Sep 17 00:00:00 2001
+From: "H. Vetinari" <h.vetinari@gmx.com>
+Date: Wed, 15 Nov 2023 10:06:16 +1100
+Subject: [PATCH 6/6] set pyarrow._hotfix_installed = True
+
+---
+ python/pyarrow/__init__.py | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/python/pyarrow/__init__.py b/python/pyarrow/__init__.py
+index ee0d07bb2..4b4ad0714 100644
+--- a/python/pyarrow/__init__.py
++++ b/python/pyarrow/__init__.py
+@@ -66,6 +66,10 @@ import pyarrow.lib as _lib
+ if _gc_enabled:
+     _gc.enable()
+ 
++# set to true because pandas unconditionally checks this attribute;
++# that code doesn't understand that we backported the relevant fix
++_hotfix_installed = True
++
```

Comment:
FWIW, I _think_ this shouldn't necessarily be needed. Even with the patch being backported, there is no harm in pandas unregistering the pyarrow extension type and registering its own version of it. 

The only reason that we check that attribute was to avoid doing the unregister/register step twice when pyarrow-hotfix was installed and imported by the user. And to ensure that `pyarrow_hotfix.uninstall()` would work as a remedy.

Reply:
I think the tests here will fail with pandas 2.1.3 if we don't have that value set, and by extensions, it will fail for users who attempt to use those paths and then tell them to install this hotfix package, which is really not good advice, especially if we've fixed it already.